app/src/main/java/com/mborowiec/antennapod/activity/MainActivity.java

```diff
@@ -490,6 +495,22 @@ private void handleNavIntent() {
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
         setIntent(intent);
+
+        handleIntent(intent);
+    }
+
+    /**
+     * Checks whether the activity was launched via a deep link.
+     *
+     * @param intent incoming intent
+     */
+    private void handleIntent(Intent intent) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            if (Objects.equals(intent.getAction(), Intent.ACTION_VIEW)) {
+                Uri uri = intent.getData();
+                if (uri != null) handleDeeplink(uri);
```

Comment:
Does the project have a style guide? If yes and if this style is allowed, go for it ;-) Otherwise, I would prefer moving it to the next line with {}